internal/lightcone/destruction/collapsingsky/collapsingsky.go

```diff
@@ -0,0 +1,63 @@
+package collapsingsky
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Check key.Modifier = "collapsing-sky"
+	Buff  key.Modifier = "collapsing-sky-buff"
+)
+
+func init() {
+	lightcone.Register(key.CollapsingSky, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        3,
+		Path:          model.Path_DESTRUCTION,
+		Promotions:    promotions,
+	})
+
+	modifier.Register(Check, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnBeforeAttack: onBeforeAttack,
+			OnAfterAttack:  onAfterAttack,
+		},
+	})
+
+	modifier.Register(Buff, modifier.Config{
+		StatusType: model.StatusType_STATUS_BUFF,
+	})
```

Comment:
Looking at the datamine:
* This should just be 1 modifier that only subscribes to `OnBeforeHit` and modifies the damage data
* This modifier is permanently attached and is not registered as a `BUFF`